packages/adblocker/src/engine/domains.ts

```diff
@@ -11,6 +11,24 @@ import { toASCII } from '../punycode.js';
 import { StaticDataView, sizeOfUint32Array, sizeOfUTF8 } from '../data-view.js';
 import { binLookup, hasUnicode, HASH_INTERNAL_MULT } from '../utils.js';
 
+export function testNetworkParts(parts: string[]): boolean {
+  for (const part of parts) {
+    if (part.length === 0) {
+      return false;
+    } else if (part.startsWith('|')) {
+      return false;
+    } else if (part.endsWith('|')) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+export function normalizeNetworkPartsLiteral(parts: string) {
```

Comment:
Let's add some comment and/or make naming more explicit. It's hard to understand what "normalizing" means, I think it's too generic.

Reply:
outdated, decided not to normalize the value on runtime.